src/ILCompiler.ReadyToRun/src/Compiler/ReadyToRunCompilerContext.cs

```diff
@@ -70,5 +72,53 @@ protected override RuntimeInterfacesAlgorithm GetRuntimeInterfacesAlgorithmForNo
         {
             return BaseTypeRuntimeInterfacesAlgorithm.Instance;
         }
+
+        private class VectorFieldLayoutAlgorithm : FieldLayoutAlgorithm
+        {
+            private FieldLayoutAlgorithm _fallbackAlgorithm;
+
+            public VectorFieldLayoutAlgorithm(FieldLayoutAlgorithm fallbackAlgorithm)
+            {
+                _fallbackAlgorithm = fallbackAlgorithm;
+            }
+
+            public override bool ComputeContainsGCPointers(DefType type)
+            {
+                return _fallbackAlgorithm.ComputeContainsGCPointers(type);
+            }
+
+            public override DefType ComputeHomogeneousFloatAggregateElementType(DefType type)
+            {
+                return _fallbackAlgorithm.ComputeHomogeneousFloatAggregateElementType(type);
+            }
+
+            public override ComputedInstanceFieldLayout ComputeInstanceLayout(DefType type, InstanceLayoutKind layoutKind)
+            {
+                List<FieldAndOffset> fieldsAndOffsets = new List<FieldAndOffset>();
+                foreach (FieldDesc field in type.GetFields())
+                {
+                    fieldsAndOffsets.Add(new FieldAndOffset(field, LayoutInt.Indeterminate));
```

Comment:
This should limit to instance fields in case people add some (didn't check if there are any).

Reply:
Fixed in 10th commit.